### DISCUSSION TITLE
Do I Need Air SDK For Flex Developers?

### DISCUSSION THREAD

**Discussion by dmunsie** [airsdk/Adobe-Runtime-Support#3001] [Q&A] [answered]
Sorry for the noob question, I'm basically starting over in the world of AS3 development and I can't remember how or why, if I need the Air SDK for Flex Developers or not. So...  how would I know which SDK to install? (The one with the Flex SDK or without.) Thx.

**Comment by ajwfrost** [comment]
The "AIR SDK for Flex Developers" is essentially a version that doesn't have the 'new' ActionScript compilers/tools that Adobe developed more recently and that don't support MXML. So, if you are using Flex, then you'd (generally) be creating MXML files as well as ActionScript, and to provide the Flex components and support for MXML you would have downloaded the Flex SDK from Apache.

If you've got that SDK, then you have the compiler etc already; you just need the AIR-specific files and frameworks to go on top of that, so you would download the "AIR SDK for Flex Developers" and then copy that over the top of the Flex SDK to result in an overall SDK that can be used for Flex development and that also has access to the newer AIR features/APIs.

If you're just using Adobe Animate, or any "pure ActionScript" project/IDE, then you should just get the normal AIR SDK which is a standalone SDK for developing, includes the (new) ActionScript compiler, but won't work with Flex/MXML.

Hope that helps to clarify it! FYI if you did want to use Flex then it may be worth checking out Apache Royale instead, which uses MXML/AS still but which has different classes/structures for the components: a lot more lightweight and performant, and with the ability to compile to SWFs or to JavaScript..

thanks

**Comment by dmunsie** [comment]
Thank you. If I use [embed] does that trigger the need for Flex? The reason I'm asking is, the only way my app compiles is if I use the Flex SDK. Just trying to better understand why that's happening. Thx.

**Reply by ajwfrost** [reply, marked as answer]
No, the `[embed]` tag is something that the normal AIR compiler copes with, it results in an asset tag being created for the resource within the SWF file. Flex should only be needed if you're using MXML or if you've got any imports from the Flex frameworks (e.g. "mx.something.class" or "spark.something.class").

Not sure what else might cause it to only work with the Flex compiler, if you're able to share the error output then we can look..

thanks

**Comment by dmunsie** [comment]
Thanks! I found the issue, I'm using the old Greensock Tweener and it indeed does use the MX library. lol..
